solr/core/src/java/org/apache/solr/schema/IndexSchema.java

```diff
@@ -1225,6 +1227,16 @@ public SchemaField getFieldOrNull(String fieldName) {
     return f;
   }
 
+  public SchemaField getFieldOrNull(SortOrder sortOrder) {
+    if (sortOrder instanceof SortField) {
```

Comment:
This is another scary hackity code!

Reply:
We can cut Solr over to using SortOrder instead of SortField as well, but that would make this PR massive, so I was hoping to do that in a followup.